### PR TITLE
Format/Re-format the drives on request

### PR DESCRIPTION
**Pull request by Praveenrajmani**
- Format the unformatted drives while adding drives with filesystem being set in the requested format
- Reformat the drive if the filesystem and force is set in the requested format

Steps to test (By directly applying the directcsi yaml)

- Set the filesystem in the requested format if the csidrive is not formatted yet. Apply and check if the drive is formatted with the requested fs
- Set the filesystem and force flag in the requested format if the csi drive is already formatted and has a filesystem. Apply and check if the drive is reformatted

Note: Do not forget to set the directcsiOwned flag to true before applying the yaml

### CODE REVIEW

**Review by nitisht — DISMISSED** [review]

File: build.sh [hunk 1]
```diff
@@ -11,3 +11,5 @@ find . | grep .go$ | xargs addlicense -check
 echo "building direct-csi $CSI_VERSION"
 CGO_ENABLED=0 go build -tags 'osusergo netgo static_build' -ldflags="-X github.com/minio/direct-csi/cmd.Version=$CSI_VERSION -extldflags=-static"
 
+apt update -y
+apt-get -y install xfsprogs
```
Comment (wlan0) [inline build.sh:15]: put this in the dockerfile

File: pkg/node/node_controller.go [hunk 2]
```diff
@@ -77,20 +73,47 @@ func (b *DirectCSIDriveListener) Update(ctx context.Context, old, new *v1alpha1.
...
+		isForceOptionSet := new.RequestedFormat.Force
+		fsType := new.RequestedFormat.Filesystem
+		copiedDrive := new.DeepCopy()
+		if fsType != "" {
+			if new.Filesystem != "" && !isForceOptionSet {
+				return status.Errorf(codes.InvalidArgument, "Need to set `force` option to override the format of this drive: %s", old.ObjectMeta.Name)
+			}
+			if new.Mountpoint != "" {
+				// Check and unmount if the drive is already mounted
+				if err := UnmountIfMounted(new.Mountpoint); err != nil {
```
Comment (wlan0) [inline pkg/node/node_controller.go:89]: we should first check if there are any volumes being used from this drive before unmounting.

**Review by wlan0 — CHANGES_REQUESTED** [review]
ensure that there are no volumes in use from a drive before unmounting it

**Comment by Praveenrajmani** [comment]
@nitisht  @wlan0 Can we get this PR merged? I can send a separate PR for purge.

**Review by nitisht — APPROVED** [review]

File: pkg/controller/controller_test.go [hunk 3]
```diff
@@ -297,3 +297,104 @@ func TestFilterDrivesByFsType(t1 *testing.T) {
 		})
 	}
 }
+
+func TestFilterDrivesByRequestedFormat(t1 *testing.T) {
```
Comment (wlan0) [inline pkg/controller/controller_test.go:301]: :+1

**Review by wlan0 — APPROVED** [review]

**Comment by Praveenrajmani** [comment]
Fixed the conflicts. Can you re-approve @wlan0 @nitisht

**Review by wlan0 — APPROVED** [review]

**Pull request merged**